packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -0,0 +1,497 @@
+    public static dynamoDbPutItem(keyName: string, valueArg?: string, idArg?: string): MappingTemplate {
+        return this.fromString(`{
+            "version" : "2017-02-28",
+            "operation" : "PutItem",
+            "key" : {
+                "${keyName}": $util.dynamodb.toDynamoDBJson(${idArg ? `$ctx.args.${idArg}` : '$util.autoId()'}),
+            },
+            "attributeValues" : $util.dynamodb.toMapValuesJson(${valueArg ? `$ctx.args.${valueArg}` : '$ctx.args'})
+        }`);
+    }
+
+    /**
+     * Mapping template to invoke a Lambda function
+     * @param payload the VTL template snippet of the payload to send to the lambda
+     */
+    public static lambdaRequest(payload: string): MappingTemplate {
+        return this.fromString(`{"version": "2017-02-28", "operation": "Invoke", "payload": ${payload}}`);
+    }
+
+    /**
+     * Mapping template to return the Lambda result to the caller
+     */
+    public static lambdaResult(): MappingTemplate {
+        return this.fromString('$util.toJson($ctx.result)');
+    }
+
+    /**
+     * this is called to render the mapping template to a VTL string
+     */
+    public abstract renderTemplate(): string;
+
+}
+
+class StringMappingTemplate extends MappingTemplate {
+
+    constructor(private readonly template: string) {
+        super();
+    }
+
+    public renderTemplate() {
+        return this.template;
+    }
+}
+
+export interface BaseResolverProps {
+    /**
+     * name of the GraphQL type this resolver is attached to
+     */
+    readonly typeName: string;
+    /**
+     * name of the GraphQL fiel din the given type this resolver is attached to
+     */
+    readonly fieldName: string;
+    /**
+     * configuration of the pipeline resolver
+     * @default - create a UNIT resolver
+     */
+    readonly pipelineConfig?: CfnResolver.PipelineConfigProperty | IResolvable;
+    /**
+     * The request mapping template for this resolver
+     * @default - No mapping template
+     */
+    readonly requestMappingTemplate?: MappingTemplate;
+    /**
+     * The response mapping template for this resolver
+     * @default - No mapping template
+     */
+    readonly responseMappingTemplate?: MappingTemplate;
+}
+
+export interface ResolverProps extends BaseResolverProps {
+    /**
+     * The API this resolver is attached to
+     */
+    readonly api: GraphQLApi;
+    /**
+     * The data source this resolver is using
+     */
+    readonly dataSource: BaseDataSource;
+}
+
+export class Resolver extends Construct {
+
+    /**
+     * the ARN of the resolver
+     */
+    public readonly arn: string;
+
+    private resolver: CfnResolver;
+
+    constructor(scope: Construct, id: string, props: ResolverProps) {
+        super(scope, id);
+
+        this.resolver = new CfnResolver(this, 'Resource', {
+            apiId: props.api.apiId,
+            typeName: props.typeName,
+            fieldName: props.fieldName,
+            dataSourceName: props.dataSource.name,
```

Comment:
Small spot... `dataSourceName` is optional - you can have resolvers with no data source

Reply:
I changed it, thanks.